app/main.py

```diff
@@ -83,33 +80,28 @@ class QuerySequence(BaseModel):
 
 
 class BlastJob(BaseModel):
-    genome_ids: list[str]
+    genome_ids: list[UUID]
     query_sequences: list[QuerySequence]
     parameters: BlastParams
 
-    @validator("genome_ids", each_item=True)  # Check each genome id
-    def check_genome_id(cls, uuid):
-        assert uuid in app.genome_ids, f"{uuid} is not a valid genome ID"
-        return uuid
-
 
 class JobIDs(BaseModel):
     job_ids: list[str]
 
 
-# Infer the target species index file for BLAST payload
-def get_blast_filename(genome_id: str, db_type: str) -> str:
-    suffix = f"{db_type}.toplevel" if db_type.startswith("dna") else f"{db_type}.all"
-    return f"ensembl/{app.genome_ids[genome_id]}/{db_type}/{app.genome_ids[genome_id]}.{suffix}"
+# Infer the target db path for BLAST job
+def get_db_path(genome_id: str, db_type: str) -> str:
+    suffix_map = {"dna": "unmasked", "dna_sm": "softmasked"} #dna/dna_sm/cdna/pep
+    return f"ensembl/{genome_id}/{suffix_map.get(db_type, db_type)}"
 
 
 # Submit a BLAST job to JD. Returns a resolvable for fetching the response.
 async def run_blast(
     query: dict, blast_payload: dict, genome_id: str, db_type: str
 ) -> dict:
     blast_payload["sequence"] = query["value"]
-    blast_payload["database"] = get_blast_filename(genome_id, db_type)
-    url = "http://www.ebi.ac.uk/Tools/services/rest/ncbiblast_ensembl/run"
+    blast_payload["database"] = get_db_path(genome_id, db_type)
+    url = "http://wwwdev.ebi.ac.uk/Tools/services/rest/ncbiblast_ensembl/run"
```

Comment:
I would avoid hardcoded string. Let the config decide based on environment which URL to use

Reply:
Updating configmap is way better than frequent changes in the code base itself in my opinion

Reply:
@andrewyatz more on this here https://12factor.net/config

Reply:
Done